### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by LOctavio**
I built the folowing methods using TDD:

- A method called factorial that takes one argument, an integer N, and returns the factorial for that number.
- A method called reverse that takes one argument, a string word, and returns word reversed.
- a method called fizzbuzz that takes one argument, an integer N, and returns a string. The returned string is constructed following these rules:
  - When N is divisible by 3, return "fizz".
  - When N is divisible by 5, return "buzz".
  - When N is divisible by 3 and 5, return "fizzbuzz".
  - Any other case, return N as a string.

### CODE REVIEW

**Review by oxenprogrammer — CHANGES_REQUESTED** [review]
Hi @LOctavio, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
- [ ] Currently you have failing tests. Please let us fix them and also write edge cases tests for all the functions. Below is a screenshot of the failing tests.

tdd-failed

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  end
+  context 'Testing the factorial method' do
+    it 'if the argumen is 0 the result is 1' do
+      expect(@solver.factorial(0)).to eq(1)
+    end
+
+    it 'if the argumen is 4 the result is 24' do
+      expect(@solver.factorial(4)).to eq(24)
+    end
+
```
Comment (oxenprogrammer) [inline spec/solver_spec.rb:15]: **Optional Test**
```
   it 'ishould raise an exception' do
      expect { @solver.factorial(-5) }.to raise_exception
    end
```

**Review by oxenprogrammer — APPROVED** [review]
Hi @LOctavio, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
